store/src/db/migrations.rs

```diff
@@ -50,16 +50,16 @@ pub static MIGRATIONS: Lazy<Migrations> = Lazy::new(|| {
             block_number INTEGER NOT NULL,
 
             PRIMARY KEY (nullifier),
+            CONSTRAINT fk_block_num FOREIGN KEY (block_number) REFERENCES block_headers (block_num),
             CONSTRAINT nullifiers_nullifier_is_digest CHECK (length(nullifier) = 32),
             CONSTRAINT nullifiers_nullifier_prefix_is_u16 CHECK (nullifier_prefix >= 0 AND nullifier_prefix < 65536),
-            CONSTRAINT nullifiers_block_number_is_u32 CHECK (block_number >= 0 AND block_number < 4294967296),
-            FOREIGN KEY (block_number) REFERENCES block_headers (block_num)
+            CONSTRAINT nullifiers_block_number_is_u32 CHECK (block_number >= 0 AND block_number < 4294967296)
         ) STRICT, WITHOUT ROWID;
         ",
     )])
 });
 
 #[test]
 fn migrations_test() {
-    assert!(MIGRATIONS.validate().is_ok());
+    assert_eq!(MIGRATIONS.validate(), Ok(()));
```

Comment:
Why is this change needed? Isn't it equivalent?

Reply:
the result of `validate()` is a bool, that erases the error message. Comparing against `Ok(())` preserver the sqlite3 error if the test fails